Mathlib/Topology/Compactness/DeltaGeneratedSpace.lean

```diff
@@ -181,3 +180,247 @@ instance Sum.deltaGeneratedSpace [DeltaGeneratedSpace X] [DeltaGeneratedSpace Y]
+    suffices h : 1 ≤ (σ t) * (2 * (2 : ℝ) ^ (n : ℝ)) by rw [mul_left_comm] at h; linarith
+    refine (mul_inv_cancel₀ ?_).symm.le.trans <| mul_le_mul_of_nonneg_left h (σ t).2.1
+    exact (coe_pos.2 ht').ne.symm⟩
+  exact γ n t'
+
+/-- On closed intervals [1 - 2 ^ n, 1 - 2 ^ (n + 1)], `sigmaConcatFun γ x` agrees with a
+  reparametrisation of `γ n`. -/
+private lemma Path.sigmaConcatFun_eqOn {X : Type*} [TopologicalSpace X] {s : ℕ → X}
+    (γ : (n : ℕ) → Path (s n) (s n.succ)) {x : X} (n : ℕ) :
+    Set.EqOn (sigmaConcatFun γ x) (fun t ↦ (γ n).extend (2 * (1 - (1 - t) * (2 ^ n))))
+    (Set.Icc (σ ⟨1 / 2 ^ n, by simp [inv_le_one₀, one_le_pow₀]⟩)
+      (σ ⟨1 / 2 ^ (n+1), by simp [inv_le_one₀, one_le_pow₀]⟩)) := fun t ht ↦ by
+  simp only [Set.mem_Icc, ← Subtype.coe_le_coe, coe_symm_eq] at ht
+  have ht' : t < 1 := coe_lt_one.1 <| ht.2.trans_lt <| by simp
+  have ht'' : 1 - t.1 > 0 := by linarith [coe_lt_one.2 ht']
+  simp only [sigmaConcatFun, ht', ↓reduceDIte, coe_symm_eq, Real.rpow_natCast]
+  by_cases hn : ⌊-Real.logb 2 (1 - ↑t)⌋₊ = n
+  · refine congr (by rw [hn]) ?_
+    rw [Set.projIcc_of_mem _ <| Set.mem_Icc.1 ⟨?_, ?_⟩]
+    · simp [hn]
+    · have h := mul_le_mul_of_nonneg_right ht.1 (a := 2 ^ n) (by simp)
+      rw [sub_mul, IsUnit.div_mul_cancel (by simp)] at h
+      linarith
+    · have h := mul_le_mul_of_nonneg_right ht.2 (a := 2 ^ (n+1)) (by simp)
+      rw [sub_mul, IsUnit.div_mul_cancel (by simp), pow_succ] at h
+      linarith
+  · replace hn : ⌊-Real.logb 2 (1 - ↑t)⌋₊ = n + 1 := by
+      refine le_antisymm ?_ <| n.succ_le_of_lt <| (Ne.symm hn).lt_of_le ?_
+      · refine (Nat.floor_le_floor <| neg_le_neg <| Real.logb_le_logb_of_le one_lt_two (by simp) <|
+          le_sub_comm.1 ht.2).trans <| by simp [- Nat.cast_add, Real.logb_pow]
+      · exact (Eq.le (by simp [Real.logb_pow])).trans <| Nat.floor_le_floor <|
+          neg_le_neg <| Real.logb_le_logb_of_le one_lt_two ht'' <| sub_le_comm.1 ht.1
+    have ht'' : (2 * (1 - (1 - t.1) * 2 ^ n)) = 1 := by
+      suffices h : t.1 = 1 - 1 / 2 ^ (n + 1) by
+        rw [h, pow_succ]; simp [mul_sub, show (2 : ℝ) - 1 = 1 by ring]
+      refine le_antisymm ht.2 ?_
+      suffices h : n + 1 ≤ -Real.logb 2 (1 - ↑t) by
+        rw [sub_le_comm]
+        convert Real.rpow_le_rpow_of_exponent_le one_le_two <| le_neg.1 h using 1
+        · rw [Real.rpow_logb two_pos one_lt_two.ne.symm ht'']
+        · rw [Real.rpow_neg two_pos.le]; simp [← Real.rpow_natCast]
+      refine Eq.trans_le (by simp [hn]) <| Nat.floor_le <| Left.nonneg_neg_iff.2 ?_
+      refine Real.logb_nonpos ?_ ?_ ?_ <;> simp [t.2.1, t.2.2]
+    rw [ht'', extend_one]; convert (γ (n + 1)).source
+    simp [hn, pow_succ]
+    linear_combination ht''
+
+/-- The concatenation of countably many paths leading up to some point `x`. -/
+private noncomputable def Path.sigmaConcat {X : Type*} [TopologicalSpace X] {s : ℕ → X}
+    (γ : (n : ℕ) → Path (s n) (s n.succ)) (x : X) {b : ℕ → Set X} (hb : (𝓝 x).HasAntitoneBasis b)
```

Comment:
It feels a bit weird to ask for an antitone basis when in fact an *eventually* antitone basis is enough. I want to say something like "the paths uniformly converge towards the point", but I'm not sure how to say that without the data of a uniformity